app/models/host.rb

```diff
@@ -99,8 +99,13 @@ class Host < ApplicationRecord
     )
   }
 
-  scope :joins_test_result_profiles, lambda {
-    left_outer_joins(:test_result_profiles)
+  scope :joins_test_result_profiles, lambda { |profile = nil|
+    profile ||= RequestStore.store['scoped_search_context_profiles']
+    profile_ids = profile&.pluck(:id) || []
```

Comment:
Can this be done by doing a separate query for ids?

Reply:
Hmm I'm not sure what you mean by this.. Did you have something in mind?

Reply:
Here we're retrieving ids from the database and then passing it into another database query. Can we have the database deal with it internally?

Reply:
Sorry, I had a typo there, I meant "without doing a separate query".

Reply:
Ah right I see what you mean now. Yeah, I can think of something.